pkg/jobs/hive/hive.go

```diff
@@ -239,13 +239,16 @@ func monitorClusterStatus(client clientv1.Client, hiveset hiveclient.Interface,
 			for _, condition := range cluster.Status.Conditions {
 				// the cluster provisioning will be treated as failed if the ClusterDeployment has any of
 				// the following conditions:
-				// 1) ProvisionFailedCondition is True, it indicates that a provision failed;
-				// 2) ProvisionStoppedCondition is True, it indicates that at least one provision attempt was
+				// 1) ProvisionStoppedCondition is True, it indicates that at least one provision attempt was
```

Comment:
To be super picky, it's not valid to assume that "at least one provision attempt was made". For example, if RequirementsMet is False, or if installAttemptsLimit is zero, we won't attempt a provision.

Reply:
Thanks @2uasimojo for the comment. If the `installAttemptsLimit` is 0, hive won't attempt a provision. And in this situation, changing `installAttemptsLimit` to a non-zero value, like 1, will not trigger the provision. So it should be treated as a failed state for curator as well.